From ce84fa12f9d9ae751da24638d4437abda26aaece Mon Sep 17 00:00:00 2001
From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Thu, 8 Feb 2018 17:50:16 +0100
Subject: [PATCH 02/27] fix full frame updates for VNC clients

RH-Author: Daniel P. Berrange <berrange@redhat.com>
Message-id: <20180208175041.5634-3-berrange@redhat.com>
Patchwork-id: 78933
O-Subject: [RHEL-7.5 qemu-kvm PATCH v1 02/27] fix full frame updates for VNC clients
Bugzilla: 1527405
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

From: Stephan Kulow <coolo@suse.de>

If the client asks for !incremental frame updates, it has lost its content
so dirty doesn't matter - it has to see the full frame, so setting force_update

Signed-off-by: Stephan Kulow <coolo@suse.de>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Reviewed-by: Peter Lieven <pl@kamp.de>
(cherry picked from commit 07535a890200e640517be0ae04fcff28860ecd37)
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 ui/vnc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/vnc.c b/ui/vnc.c
index 29b216c..5226295 100644
--- a/ui/vnc.c
+++ b/ui/vnc.c
@@ -1777,6 +1777,7 @@ static void framebuffer_update_request(VncState *vs, int incremental,
         return;
     }
 
+    vs->force_update = 1;
     vnc_set_area_dirty(vs->dirty, width, height, x, y, w, h);
 }
 
-- 
1.8.3.1

